src/runner.ts

```diff
@@ -102,155 +104,160 @@ export interface IRunnerOptions {
     version?: boolean;
 }
 
-export class Runner {
-    private static trimSingleQuotes(str: string) {
-        return str.replace(/^'|'$/g, "");
-    }
-
-    constructor(private options: IRunnerOptions, private outputStream: NodeJS.WritableStream) { }
+export const enum Status {
+    Ok = 0,
+    FatalError = 1,
+    LintError = 2,
+}
 
-    public run(onComplete: (status: number) => void) {
-        if (this.options.version) {
-            this.outputStream.write(Linter.VERSION + "\n");
-            return onComplete(0);
+export async function run(options: Options, outputStream: NodeJS.WritableStream): Promise<Status> {
```

Comment:
While you're in the area, would it make sense to stop using any `console` messages at all? In addition to passing the CLI `options` and an `outputStream` it makes sense to also pass an `errorStream`.